memo_app.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'cgi'
+require 'json'
+
+set :environment, :production
+
+class MemoData
+  attr_accessor :memos, :id
+
+  def initialize
+    @memos = {}
+    @id = 0
```

Comment:
(1) memos の初期化
52行目、`MemoData.new` したあとに `#read_memo` を呼んでいますね。必ず呼ぶ必要があるのでしたら、初期化の時点で呼んでしまってもよいかな？と思いました。

(2) `@id` のインスタンス変数
`@id` は `add_memo` で新しいメモを追加するときのみ使用しますよね。そうであれば、常に保持している必要がないように思います。
なのでインスタンス変数で持たないようにするとよさそうです。

以上2点の理由から、ここの処理はこう書き換えてもよさそうです。

```suggestion
    @memo = read_memo
```